veeqtor/API-Trisixty
# #164350492 Deleting a product

#### What does this PR do?

- Implements the deleting and restoring soft-deleted product endpoints.

#### Description of Task to be completed?

As a verified and logged in business account owner, I should be able to soft delete a vendor by making a `DELETE` request to this route `/products/<product_id>/`,  be able to permanently delete a vendor with by making a `DELETE` request to this route `products/<product_id>/delete`, and also restore a soft-deleted  vendor by making a `GET` request to this route `products/<product_id>/restore`

#### How should this be manually tested?
- Fetch the branch with `git fetch origin ft-delete-a-product-164350492`
- Checkout to the branch with `git checkout ft-delete-a-product-164350492`
- Activate the virtual environment and start the app with `pipenv shell && python manage.py runserver`
- Make a DELETE request to the route `/products/<product_id>/`
- Make a DELETE request to the route `/products/<product_id>/delete/`
- Make a GET request to the route `/products/<product_id>/restore/`

`DELETE /products/<product_id>`
RESPONSE
`200 OK`

Response data should have the following structure
```
{
    "status": "success",
    "message": "Product successfully deleted."
}
```

`GET /products/<product_id>/restore`
RESPONSE
`200 OK`

Response data should have the following structure
```
{
    "status": "success",
    "message": "Product with name: "<title_name>" successfully restored.",
    "data": {
        "id": "<product_id>",
         ...
    }
}
```

`DELETE /products/<products_id>/delete`

RESPONSE
`200 OK`

Response data should have the following structure
```
{
    "status": "success",
    "message": "Product permanently deleted."
}
```
#### Any background context you want to provide?
- N/A

#### What are the relevant pivotal tracker stories?
- [#164350492](https://www.pivotaltracker.com/story/show/164350492)

#### Screenshots (if appropriate)
- N/A

#### Questions:
 - N/A
